src/Bus/CharacterBus.php

```diff
@@ -94,84 +92,72 @@ public function __construct(RefreshToken $token)
-        ])->dispatch($this->token);
-
         // Character
-        Info::dispatch($this->token->character_id);
-        AgentsResearch::dispatch($this->token);
-        Blueprints::dispatch($this->token);
-        CorporationHistory::dispatch($this->token->character_id);
-        Fatigue::dispatch($this->token);
-        Medals::dispatch($this->token);
-        Roles::dispatch($this->token);
-        Standings::dispatch($this->token);
-        Titles::dispatch($this->token);
-
-        // Clones
-        Clones::withChain([
-            new Implants($this->token),
-        ])->dispatch($this->token);
-
-        // Contacts
-        Contacts::withChain([
-            new ContactLabels($this->token),
-        ])->dispatch($this->token);
-
-        // Contracts
-        Contracts::dispatch($this->token);
-
-        // Fittings
-        Fittings::dispatch($this->token);
-
-        // Industry
-        Jobs::dispatch($this->token);
-        Mining::dispatch($this->token);
-
-        // Killmails
-        Recent::dispatch($this->token);
-
-        // Location
-        Location::dispatch($this->token);
-        Online::dispatch($this->token);
-        Ship::dispatch($this->token);
-
-        // Mail
-        Mails::withChain([
-            new MailLabels($this->token),
-        ])->dispatch($this->token);
-        MailingLists::dispatch($this->token);
-
-        // Market
-        Orders::dispatch($this->token);
-
-        // Planetary Interactions
-        Planets::dispatch($this->token);
-
-        // Skills
-        Attributes::dispatch($this->token);
-        Queue::dispatch($this->token);
-        Skills::dispatch($this->token);
-
-        // Structures
-        CharacterStructures::dispatch($this->token);
-
-        // Wallet
-        Balance::dispatch($this->token);
-        Journal::dispatch($this->token);
-        Transactions::dispatch($this->token);
+        Info::withChain([
+            // collect information related to current character state
+            new CorporationHistory($this->token->character_id),
+            new Roles($this->token),
+            new Titles($this->token),
+            (new Clones($this->token))->chain([
+                new Implants($this->token),
+            ]),
+
+            (new Location($this->token))->chain([
+                new Online($this->token),
+                new Ship($this->token),
+            ]),
+
+            (new Attributes($this->token))->chain([
+                new Queue($this->token),
+                new Skills($this->token),
+            ]),
+
+            // collect military informations
+            new Fittings($this->token),
+            new Recent($this->token),
+
+            new Fatigue($this->token),
+            new Medals($this->token),
+
+            // collect industrial informations
+            (new Blueprints($this->token))->chain([
+                new Jobs($this->token),
+                new Mining($this->token),
+                new AgentsResearch($this->token),
```

Comment:
Is there a reason why Mining and AgentsResearch is placed underneath Blueprints?

Reply:
not really, it's an industrial stack.
we can list them at same level - both work.